### DISCUSSION TITLE
cmake or g++ command to compile vulkan or opengl test cases

### DISCUSSION THREAD

**Discussion by HarrisonLeece** [ocornut/imgui#5265] [New Users Build/Link/Run issues ONLY!]
Hopefully someone here has a solution already on their computer, because stackoverflow was not so useful.

I am new to c/c++ and linux, but not so new to certain programming tasks.  I also have not had a teacher, so I know how to do certain weird things, but don't know how to get my own fish for tasks some might presume to be simple.  

I have a back-end written for a c++ program which is working, and I also want to push myself towards higher programming tasks, so imgui it is.  I have opengl, vulkan, glfw static and shared object libraries on my computer, but I am struggling to translate the included .bat files to g++ command.  

There also seems to be some  libraries which are included in the bat file which might be windows specific, and instead of asking 20 individual questions, I was hoping someone could share their solution so I can reverse engineer what is going on.

Thanks

**Comment by rokups** [comment]
You did not post information of any error. Sounds like this is not imgui problem. Please refer to other resources on how to use g++ or cmake. If you have a specific issue with building imgui, please post relevant logs and information.

**Comment by ghost** [comment]
Use a 
```
g++ main.cpp imgui/*.cpp 
```

Else printed error to use all .cpp files 
Example:

```
g++ main.cpp imgui/imgui.cpp imgui/imgui_widgets.cpp 
```
And other
:)

**Reply by rokups** [reply]
This will fail due to include paths not being set up.

**Reply by ghost** [reply]
In my menu full work))
